raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -454,6 +455,7 @@ public void initialize(
             nodeId,
             nodeDirectoryId,
             partitionState::lastVoterSet,
+            partitionState::lastVoterSetOffset,
```

Comment:
It is unfortunate that we need to add a parameter to the constructor just so that it can pass this value to `LeaderState`. I am starting to think that `QuorumState` should just have access to the `partitionState` (`KRaftControlRecordStateMachine`) object.